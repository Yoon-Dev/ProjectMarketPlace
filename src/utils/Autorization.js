import React from 'react';

export class Autorization extends React.Component{

    constructor(props){
        super(props)
            this.state = {
                isLogged: false,
                toLogin: () => {this.setState({isLogged: !this.state.isLogged})},
                userName: null,
                setUserName: username => {this.setState({userName: username})},
                nbr_article: 0,
                setNbr_article: () => {this.setState({nbr_article: this.state.nbr_article+1})}
            }
        }    

    render(){
        return(
            <AutorizationContext.Provider value={this.state}>
                {this.props.children}
            </AutorizationContext.Provider>
        )
    }
}

export const auth = new Autorization()//<Autorization/>

export const AutorizationContext = React.createContext(null)


export const withAutorization = Component => {
    const NewComponent = props => {
        return(
            <AutorizationContext.Consumer>
                { value => <Component {... value} {... props}/>}
            </AutorizationContext.Consumer>
        )
    }
    return NewComponent
}
