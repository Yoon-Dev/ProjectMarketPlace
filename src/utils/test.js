import { act } from "react-dom/test-utils";
import { fireEvent } from "@testing-library/react";
import { Breadcrumbs } from "@material-ui/core";



export function testClick(el) {
    act( () => {
      el.dispatchEvent(
        new MouseEvent("click", { bubbles: true })
      );
    });
}

export function singIn(idvalue='bru', passvalue='pass'){
  const identifiant = document.querySelector("[name='identifiant']")
  const pass = document.querySelector("[name='pass']")
  const btn = document.querySelector("[data-testid='btn-sign']")


  fireEvent.change(identifiant, {target: { value: idvalue } })
  fireEvent.change(pass, {target: { value: passvalue } })
  act( () => {
    btn.dispatchEvent(
      new MouseEvent("click", { bubbles: true })
    );
  });
}
