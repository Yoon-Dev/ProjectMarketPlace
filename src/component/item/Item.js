import React, { useState, useEffect } from "react";
import Grid from '@material-ui/core/Grid';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import AddCircleIcon from '@material-ui/icons/AddCircle';
import DeleteIcon from '@material-ui/icons/Delete';
import './Item.css'
import { compose } from 'recompose'
import { withAutorization } from '../../utils/Autorization';

const Item = props => {

    
    const useStyles = makeStyles( () => createStyles({
        center: {
            display: "flex",
            justifyContent: "center",
            marginBottom: "1.33em"
        },
        mb:{
            marginTop: "2em"
        },
        el: {
            textAlign: "center"
        },
        img: {
            maxWidth: "100%"
        }

      }));
    const style = useStyles();

    const [purchased, setPurchased] = useState(props.determinePurchased(props.id));
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°° 
      useEffect(() => {
        const self = document.getElementById(props.id)
          if(purchased){
            self.firstChild.style.background = "url('https://media.giphy.com/media/l0OXXpl20sY9G0uJy/source.gif')";
          }
      }, [purchased, props.id]);
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°° 
// send a request to delete the ressouce in BDD
    const handleSubmit = () => {
        const self = document.getElementById(props.id)
        if(purchased){
            // remove
            // props.setPurchased_article(props.id)
            props.removePurchased_article(props.tab_purchased_article.indexOf(props.id))
            props.setNbr_article(props.nbr_article -1)
            self.firstChild.style.background = "white";
        }else{
            props.setPurchased_article(props.id)
            props.setNbr_article(props.nbr_article + 1)
            self.firstChild.style.background = "url('https://media.giphy.com/media/l0OXXpl20sY9G0uJy/source.gif')";
        }
        setPurchased(!purchased)
    }
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°
// °°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°°° 
    return(
    <Grid id={props.id} className={style.center} item xs={6}>
        <div className="card">
            <Grid container justify="center" alignItems="center" className={style.mb}>
                <Grid item xs={12} className={style.center}> 
                    <h4 className={style.el}>{props.nom}</h4> 
                </Grid>
                <Grid item xs={12} className={style.center}> 
                    <img src="https://upload.wikimedia.org/wikipedia/en/4/46/IMG_Academy_Logo.jpg" alt="product" className={style.img}/>
                </Grid>
                <Grid item xs={12} className={style.center}> 
                    <Button variant="contained" color={ purchased ? "secondary" : "primary" } startIcon={ purchased ? <DeleteIcon /> : <AddCircleIcon/>}className={style.el} onClick={handleSubmit}>
                        { purchased ? "Enlever du panier" : "Acheter" }
                    </Button>
                </Grid>
            </Grid>
        </div>
        
    </Grid>
    )
}

export default compose(withAutorization)(Item);