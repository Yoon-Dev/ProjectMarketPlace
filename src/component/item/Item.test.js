import React from 'react';
import ReactDOM from 'react-dom';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { testClick, singIn } from '../../utils/test.js';
import pretty from "pretty";
import renderer from 'react-test-renderer';
import Item from './Item';
import { Autorization } from '../../utils/Autorization.js';

let container = null;

beforeEach( () => {
  container = document.createElement("div");
  document.body.appendChild(container);
})


afterEach( () => {
  unmountComponentAtNode(container);
  container.remove();
});



test("it display data correctly", () => {
  act(() => {
    render(      
    <Autorization>
      <Item key={1} id={1} nom="item test" url="https://jestjs.io/docs/en/tutorial-react" />
    </Autorization>, container);
  });
    const name = document.querySelector("[data-testid='item-name']")
    const id = document.querySelector("[data-testid='item-id']")
    const img = document.querySelector("[data-testid='item-img']")
    // get the correct name
    expect(name.textContent).toBe("item test")
    // correct id
    expect(id.getAttribute('id')).toBe("1")
    //  correct url
    expect(img.getAttribute('src')).toBe("https://jestjs.io/docs/en/tutorial-react")

  });

test("it match snapshot", () => {

    const component = renderer.create(
      <Autorization>
        <Item key={1} id={1} nom="item test" url="https://jestjs.io/docs/en/tutorial-react" />
      </Autorization>
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
