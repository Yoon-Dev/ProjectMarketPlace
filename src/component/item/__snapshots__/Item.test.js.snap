// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`it match snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-center-159 MuiGrid-item MuiGrid-grid-xs-3"
  data-testclass="item"
  data-testid="item-id"
  id={1}
>
  <div
    className="card"
  >
    <div
      className="MuiGrid-root makeStyles-mb-160 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-center-159 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          className="makeStyles-el-161"
          data-testid="item-name"
        >
          item test
        </h4>
      </div>
      <div
        className="MuiGrid-root makeStyles-center-159 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="product"
          className="makeStyles-img-162"
          data-testid="item-img"
          src="https://jestjs.io/docs/en/tutorial-react"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-center-159 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-el-161 MuiButton-containedPrimary"
          data-testclass="btn-items"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
            </span>
            Acheter
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
